### ISSUE TITLE
FR: Is there any method to change a specific font?

### ISSUE THREAD

**Issue by devKyanon** [koreader/koreader#10969]
Hello, thank you for developing this amazing app.
I really enjoy reading books with Koreader :)

I'd like to ask how to change a SPECIFIC font of a epub file.

I am likely reading korean epubs but some korean fonts are not recognized well in Koreader.
Hence, opening those files with a default font looking ugly or I should globally change all fonts to another one. 

I think I can manage this issue if I can specific a font list in a epub one-by-one manually, but I cannot find the way.
I tried many ways, unfortunately, a font-family option doesn't work as I want to, and it is limited to Korean.

Moon reader provides the option to replace a specific font. 
In short, I simply ask that Koreader has the similar option.
I hope to feel the publishers' intend and design principles when I read books.

**Comment by poire-z** [comment]
> some korean fonts are not recognized well in Koreader
> I can specific a font list in a epub one-by-one manually, but I cannot find the way.

**Comment by poire-z** [comment]
> some korean fonts are not recognized well in Koreader
> I can specific a font list in a epub one-by-one manually, but I cannot find the way.
> Moon reader provides the option to replace a specific font.

Not sure what you mean with "replacing a specific font", and dunno how the UI for that looks in MoonReader.

If you mean whenever the engine would want to use a font named Georgia, you would like to have it use Verdana, then no, this is not possible.
Also, our "Font-family fonts>" stuff requires the EPUB to have CSS specifying `font-family: classicfamilyname` in its stylesheet; and there's no real reason for publishers to do that.

If they do specify `P.koreanText { font-family: "SomeKoreanEmbeddedFont.ttf" };` we should use that. It it does not work well, we're ready to investigate if you can provide some sample - unless it's just that you don't like that SomeKoreanEmbeddedFont.ttf font :)

What you can do:
- Bottom menu, last tab: `Embedded font off` so no embedded font is used, and you main KOReader fonts are used instead
- Top menu, `Style tweaks > Text > Ignore publisher font families` - so no font-family is used, even if they specify a font that you happen to have.
- Top menu, `Fonts >` and just set your preferred Korean font.
- or if you need a good Latin/English font, set it, and set you preferred Korean font as the first fallback with long-press on it.
- you could also create a user style tweak with `* { font-family: "MyPreferedKoreanFont" !important }` that you could enable when needed when reading Korean books.

(This was tedious to write, I kept writting `KORean` :)

**Comment by hugleo** [comment]
Tips like that should be added to a wiki so they can't be lost and rewritten again.

**Comment by pazos** [comment]
> Tips like that should be added to a wiki so they can't be lost and rewritten again.

Agree. Maybe it deserves it's own page "Font support"?, ie: expanding https://github.com/koreader/koreader/wiki/Fonts

I would volunteer but I'm gladly unaware about the topic :p. 

Maybe, if nobody beats me, I can copy chunks of the text above and trust the free software magic sauce that happens here?

**Comment by devKyanon** [comment]
Thank you @poire-z!
Yeap, I can understand what was tedious. At I first heard of KOreader, it was for Korean reader :p  

I am going to dig in the style tweaks! I feel your suggestions may be the way I am looking for.

And, here, I attached a link for a screenshot of Moon-reader for changing fonts to clarify my intent
(Since it is Korean, I put on some description)

https://imgur.com/a/RqcZINV

It would be great if Koreader offered this option.

Anyway, Koreader is really good, I think it's the best ebook reader, thanks as always!

**Comment by Frenzie** [comment]
You can upload images straight on GitHub btw. ^_^ 

<img src=https://github.com/koreader/koreader/assets/202757/a66ec28a-e52b-4557-8f50-7182936fdbfe width=60%>
